适配安卓模拟器

antecanis8 opened the pull request:
TODO LIST:
- [x] 模拟器控制类，截屏和操控，实现自动战斗
- [ ] 控制器中缺少向下滚动 按住拖拽
- [x] 完善多分辨率适配
- [ ] 战斗场景外，模拟器使用的模板需要适配
- [x] 移除来自alas的多余代码
- [x] 将模拟器config从alas更改到one-dragon
- [x] one-dragon拉起模拟器和游戏
- [x] 前端：设置中完善模拟器config相关
- [ ] 生成新的requirement.txt





```
战斗画面  头像3-1 3-2 3-3 普通攻击 交互键

大世界 信息 地图 

菜单 返回
```

antecanis8 committed:
feat: 添加模拟器控制器

antecanis8 committed:
feat: 添加模拟器控制器

antecanis8 committed:
fix: 合并更新

antecanis8 committed:
feat: 完善模拟器控制器

/src/deploy以及/src/src/one_dragon/module来自LmeSzinc/StarRailCopilot
修改画面、模板相关
设置界面添加游戏平台选择框
为模板、画面相关类添加模拟器坐标

antecanis8 committed:
解决冲突

yokuminto commented:
一条龙可以用英文简称oni了

daimengdewo commented:
> 一条龙可以用英文简称oni了

确实

antecanis8 committed:
同步代码，修复GUI冲突

antecanis8 committed:
Merge remote-tracking branch 'refs/remotes/upstream/main' into emulator

# Conflicts:
#	src/one_dragon/base/operation/one_dragon_context.py
#	src/one_dragon/base/screen/screen_area.py
#	src/one_dragon/base/screen/screen_loader.py

antecanis8 committed:
feat: 为角色状态模版添加模拟器支持

现在每个模版都可以设置在模拟器内的坐标
同步上游最新的代码
红黄光识别部分需要从RGB转换成BGR才能工作，可能存在BUG

antecanis8 committed:
feat: 角色状态适配: 青衣-电压, 朱鸢-子弹数

antecanis8 committed:
feat: 快捷手册的截图坐标适配

xuzheliang135 commented:
@antecanis8兄弟，你这个是把alas的代码搬过来了？这个工程量不小啊。你有大体的时间表没？我之前用alas发现scrcpy只支持到1.2版本，没办法用新的new-display特性在真机上跑，还准备做更新来着

antecanis8 commented:
> @antecanis8兄弟，你这个是把alas的代码搬过来了？这个工程量不小啊。你有大体的时间表没？我之前用alas发现scrcpy只支持到1.2版本，没办法用新的new-display特性在真机上跑，还准备做更新来着

最近比较忙，预计1月10号之后才能投入时间来做。现在搬过来的alas代码还有比较多的不必要的，后面还得精简。

antecanis8 commented:
> > > @antecanis8兄弟，你这个是把alas的代码搬过来了？这个工程量不小啊。你有大体的时间表没？我之前用alas发现scrcpy只支持到1.2版本，没办法用新的new-display特性在真机上跑，还准备做更新来着
> > 
> > 
> > 最近比较忙，预计1月10号之后才能投入时间来做。现在搬过来的alas代码还有比较多的不必要的，后面还得精简。
> 
> 我还没仔细看一条龙的代码，之前想的是直接对获取图像和传递操作的函数重写一下，让其可以根据选项使用不同的通路（windows模拟、scrcpy模拟），应该比直接把alas搬过来要简单一些。这个想法是有什么问题吗？以至于你需要直接把alas的代码搬过来

对，是这个思路，目前已经把windows控制器和模拟器控制器做好了，截屏和操作都是可用的。用alas的好处是有几个截屏和触控的方案自动选择，包括mumu的专有截图，以及可以自动打开模拟器，自动重连等等。
现在遇到的最大的工作量是电脑端和手机端的ui界面不同，需要给手机端写截图区域坐标，而且要写的精准。这个做完就算功能上没问题了，然后是精简搬过来的alas代码，模拟器端口号等配置从alas改成oni的。

xuzheliang135 commented:
> > > > @antecanis8兄弟，你这个是把alas的代码搬过来了？这个工程量不小啊。你有大体的时间表没？我之前用alas发现scrcpy只支持到1.2版本，没办法用新的new-display特性在真机上跑，还准备做更新来着
> > > 
> > > 
> > > 最近比较忙，预计1月10号之后才能投入时间来做。现在搬过来的alas代码还有比较多的不必要的，后面还得精简。
> > 
> > 
> > 我还没仔细看一条龙的代码，之前想的是直接对获取图像和传递操作的函数重写一下，让其可以根据选项使用不同的通路（windows模拟、scrcpy模拟），应该比直接把alas搬过来要简单一些。这个想法是有什么问题吗？以至于你需要直接把alas的代码搬过来
> 
> 对，是这个思路，目前已经把windows控制器和模拟器控制器做好了，截屏和操作都是可用的。用alas的好处是有几个截屏和触控的方案自动选择，包括mumu的专有截图，以及可以自动打开模拟器，自动重连等等。 现在遇到的最大的工作量是电脑端和手机端的ui界面不同，需要给手机端写截图区域坐标，而且要写的精准。这个做完就算功能上没问题了，然后是精简搬过来的alas代码，模拟器端口号等配置从alas改成oni的。

明白了那我先写着我的scrcpy升级吧，到时候再说怎么合进去，不过你知道为什么alas要使用多种截图方式吗？scrcpy一把梭不好吗，还省得为每一种模拟器做适配

CLX20070805 commented:
感谢你
thanks you

antecanis8 committed:
同步上游代码

antecanis8 committed:
Merge remote-tracking branch 'refs/remotes/upstream/main' into emulator

# Conflicts:
#	src/one_dragon/base/screen/screen_info.py

antecanis8 committed:
feat(添加模拟器相关设置界面，移动模拟器配置文件路径):

antecanis8 committed:
Merge remote-tracking branch 'refs/remotes/upstream/main' into emulator

# Conflicts:
#	src/one_dragon_qt/view/devtools/devtools_template_helper_interface.py
#	src/zzz_od/config/game_config.py
#	src/zzz_od/context/zzz_context.py
#	src/zzz_od/gui/view/setting/app_setting_interface.py
#	src/zzz_od/gui/view/setting/setting_game_interface.py
#	src/zzz_od/operation/enter_game/open_and_enter_game.py
#	src/zzz_od/operation/enter_game/open_game.py

antecanis8 committed:
feat: 模拟器相关设置项移动到 账户管理

回滚了Yaml_config_adapter
新增了Json_config_adapter
模拟器设置 改成使用Json保存

antecanis8 committed:
feat: 添加多实例支持并记录实例索引

idk500 commented:
加油啊

georgeJzzz commented:
加油

antecanis8 committed:
Merge remote-tracking branch 'upstream/main' into emulator

antecanis8 committed:
fix: GameConfig类导入YamlConfig

antecanis8 committed:
Merge remote-tracking branch 'upstream/main' into emulator

antecanis8 committed:
refactor: 移除来自SRC的/src/deploy

antecanis8 committed:
fix: 修复模拟器平台初始化逻辑，优化组件加载

修复ScreenContext, TemplateLoader, TemplateMatcher三个类在初始化时，无法获取到正确的 platform，只能用一个 'Emulator' 作为占位符的问题
改成延迟初始化

antecanis8 committed:
feat: 支持zzz-od拉起模拟器和游戏

注意game_config里面的platform选项已经废弃

antecanis8 committed:
feat: 延迟初始化模拟器控制器

antecanis8 committed:
fix: 修复is_game_window_ready缺失的修饰器
